Lib/multiprocessing/pool.py

```diff
@@ -671,7 +691,8 @@ def join(self):
     def _help_stuff_finish(inqueue, task_handler, size):
         # task_handler may be blocked trying to put items on inqueue
         util.debug('removing tasks from inqueue until task handler finished')
-        inqueue._rlock.acquire()
+        if inqueue._reader.poll():
+            inqueue._rlock.acquire()
```

Comment:
If the `poll()` returns False, that means we should exit this function, right? Given the next part used to be done under lock, it hardly seems right that now we can do it without the lock (but maybe it's all been cleaned up here?)